features/dd-sdk-android-trace/src/test/kotlin/com/datadog/opentracing/SpanCleanerTest.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.opentracing
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.opentracing.PendingTrace.SpanCleaner
+import com.datadog.tools.unit.createInstance
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assumptions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.quality.Strictness
+import java.math.BigInteger
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+internal class SpanCleanerTest {
+
+    private var testedCleaner: SpanCleaner = SpanCleaner()
+
+    // region cleanup
+
+    @Test
+    fun `M remove random traces W add and remove`(forge: Forge) {
+        // Given
+        val pendingTraces = forge.aList(size = forge.anInt(min = 10, max = 50)) { mock<PendingTrace>() }
+        pendingTraces.forEach {
+            testedCleaner.addPendingTrace(it)
+        }
+        Assumptions.assumeTrue(testedCleaner.pendingTraces.values.size == pendingTraces.size)
+        Assumptions.assumeTrue(testedCleaner.pendingTraces.values.containsAll(pendingTraces))
+
+        // When
+        pendingTraces.shuffled().forEach {
```

Comment:
```suggestion
        pendingTraces.shuffled(Random(forge.seed)).forEach {
```

but what is the need to call `shuffled` anyway? to have a different order between `addPendingTrace` and `removePendingTrace` calls?

Reply:
yup